arch/arm/core/aarch64/reset.S

```diff
@@ -65,39 +121,21 @@ SECTION_SUBSEC_FUNC(TEXT,_reset_section,__start)
 	adr	x0, 1f
 	msr	elr_el3, x0
 	eret
-2:
-	/* Boot from EL2 not supported */
-	bl	.
-1:
 #endif
-	/* Setup vector table */
-	adr	x0, _vector_table
-
-	switch_el x1, 3f, 2f, 1f
-3:
-	/* Initialize VBAR */
-	msr	vbar_el3, x0
-
-	/* SError, IRQ and FIQ routing enablement in EL3 */
-	mrs	x0, scr_el3
-	orr	x0, x0, #(SCR_EL3_IRQ | SCR_EL3_FIQ | SCR_EL3_EA)
-	msr	scr_el3, x0
-
-	/* Disable access trapping in EL3 for NEON/FP */
-	msr	cptr_el3, xzr
-
 	/*
 	 * Enable the instruction cache, stack pointer and data access
-	 * alignment checks and disable speculative loads.
+	 * alignment checks.
 	 */
 	mov	x1, #(SCTLR_I_BIT | SCTLR_A_BIT | SCTLR_SA_BIT)
```

Comment:
Based on the experiments that I did earlier, in conjunction with enabling MMU,  we need to disable data access alignment checks by clearing `SCTLR_A_BIT` to enable unaligned access to Normal memory.

Reply:
Sure. But I thought It will be too much functionality change in one commit. Since you brought this up I will add a small change on top.

Reply:
update: while I was at it I see I could actually set SCTLR_EL1 to 'RES1' in EL3 instead of doing a read-modify-write , which is wrong in el3 context anyway as core reset to EL3. That way as per armv8.0 SCTLR.A is '0'.  So default behavior is 'Unaligned access allowed to normal memory" . If anyone wants to trap or disable unaligned access for some reason then there can be a commit to set SCTLR.A.

Reply:
I see the changes look fine now, `SCTLR_EL1` is directly set to `RES1` and `SCTLR.A` is not getting set in `SCTLR_EL1`.